src/promptflow-devkit/promptflow/_sdk/_utilities/general_utils.py

```diff
@@ -1006,8 +1006,16 @@ def create_temp_flex_flow_yaml_core(entry: Union[str, PathLike, Callable], code:
         logger.warning(f"Found existing {flow_yaml_path.as_posix()}, will not respect it in runtime.")
         with open(flow_yaml_path, "r", encoding=DEFAULT_ENCODING) as f:
             existing_content = f.read()
-    if not is_local_module(entry_string=entry, code=code):
-        logger.debug(f"Entry {entry} is not found in local, it's snapshot will be empty.")
+
+    create_yaml_in_temp_dir = False
+    if is_eval_batch_run():
```

Comment:
Can flex flow yaml be created in temp folder always ?

Reply:
Synced with @D-W- on this. By default, PF wants the entry file to be generated under the flow directory so that user code can be part of the snapshot for general cases. To avoid affecting existing behavior, we decided to use a flag for the eval batch run.

Reply:
How do we determine what is the flow directory ?